### ISSUE TITLE
Nested Entity Insertion

### ISSUE THREAD

**Issue by DavidGrinberg** [vincentlauvlwj/Ktorm#102]
Given the following pseudocode - 

```
val employee = Employee{
     //Info for a new employee
    department = Department{
        //Info for a new department
    }
  }
database.sequenceOf(Employee, withReferences = true)
    .add(employee)
```

I would expect that _both_ the department and the employee get added, however only the employee gets added and the department on the new employee remains null.

Does Ktorm support this? If not, what is the expected way to do this. I would prefer to not have to manually disentangle the dependency graph and create objects one by one myself. 

I also seem to be under a misconception of what the `withReferences` flag does here, can someone explain this?

**Comment by DavidGrinberg** [comment]
Did a bit more digging and found [this](https://github.com/vincentlauvlwj/Ktorm/issues/40) somewhat related issue, which points to [this](https://github.com/vincentlauvlwj/Ktorm/blob/master/ktorm-core/src/main/kotlin/me/liuwj/ktorm/schema/BaseTable.kt#L313-L359) code. It seems that `withReferences` flag is really only used for when you are creating entities read _from_ the DB, not putting the entities you create _into_ the DB (please correct me if I'm wrong). 

So the question remains - if I have nested entities is there a way to make Ktorm smart enough to insert both at the same time, or do I have to manually do it one by one?

**Comment by vincentlauvlwj** [comment]
> Did a bit more digging and found this somewhat related issue, which points to this code. It seems that withReferences flag is really only used for when you are creating entities read from the DB, not putting the entities you create into the DB (please correct me if I'm wrong).

Yes, you are right. The `withReferences` flag is ONLY used when we are creating entities from the DB. So such code like `database.sequence(Employees, withReferences = true).add(employee)` dose cause some misunderstandings. I will redesign the API to eliminate this misconception in the future.

**Comment by vincentlauvlwj** [comment]
> So the question remains - if I have nested entities is there a way to make Ktorm smart enough to insert both at the same time, or do I have to manually do it one by one?

Yes, you have to manually do it one by one. Ktorm doesn't support nested entity insertions now. I will also work on this in the future.
